### PR TITLE
Make the .pub files hex formatted

### PR DESCRIPTION
**Pull request by garbageslam**
This was discussed in review of https://github.com/mobilecoinfoundation/mobilecoin/pull/545

It might make these files a little user-friendlier, and if we have
other tools that use hex-encoded protobuf public address then it
seems reasonable to unify

### CODE REVIEW

**Comment by eranrund** [comment]
What is the use of this hexencoded protobuf stuff? Why isn't JSON a good alternative?

**Comment by garbageslam** [comment]
Protobuf is a fine format. The protobuf schema for public addresses is in external.proto. We seem to typically use protobuf for serialization rather than JSON generally. There are size and speed advantages. I think it leads to less boilerplate when deserializing.

The hex encoding part, I'm not sure of. It could easily be base64 or base58 on top of protobuf as well. But we have other software that are using hex encoded protobuf public address apparently, and it makes sense to unify rather than create conversion tools. I'm not sure why they chose hex.

When we made these key files protobuf, they were earlier CBOR, and before that, messagepack. When we started trying to get rid of serde, there was some reason that AccountKey could not implement Message, I can't remember now, so the private key files became json at that time.

I think having these files is good and it helps a bunch of test infra to have tooling for this. I don't care very much what the format of the files is -- if you think they should both be json or both be hex-encoded protobuf we can probably fix that. Honestly the formats of these files have pretty much been placeholder / figure it out as we go.

**Comment by garbageslam** [comment]
I guess that both of the formats are somewhat annoying operationally right now:
- the `.json` files contain root entropy as bytes, but it is a list of json integers 0-255 rather than hex encoded
- the `.pub` files are binary and can't be opened and copy pasted, but many tools want to be able to pass stuff like this on command line or paste it into a window, so even though we typically serialize things as protobuf, that turns out not to be the most convenient thing for some workflows

File: util/keyfile/src/lib.rs [hunk 1]
```diff
@@ -66,6 +68,10 @@ fn prost_to_io_error(err: mc_util_serial::DecodeError) -> std::io::Error {
     std::io::Error::new(std::io::ErrorKind::Other, Box::new(ProstError { err }))
 }
 
+fn hex_to_io_error(err: FromHexError) -> std::io::Error {
+    std::io::Error::new(std::io::ErrorKind::Other, Box::new(err))
+}
```
Comment (jcape) [inline util/keyfile/src/lib.rs:73]: Stuff like this, and the functions above it (`to_io_error`, `prost_to_io_error`, etc) and the `ProstError` type below mean we should just replace this with a `KeyFileError` that implements `From<IoError>`, `From<DecodeError>`, and `From<FromHexError>`, like we do literally everywhere else :-)

**Review by jcape — CHANGES_REQUESTED** [review]
We should take the opportunity to clean up the error handling here and use `displaydoc`.

**Comment by garbageslam** [comment]
@jcape do you have an opinion about what should ideally happen with both of these file formats? (the json root entropy files, and the pub files)

we could make them both `json` with byte members represented as hex strings?

or make them both protobuf / hex protobuf?

**Comment by jcape** [comment]
I think we should just use binary protobuf. None of the data is human readable, let along human editable, so JSON doesn't buy you anything.

**Comment by garbageslam** [comment]
ok, i guess i'm going to close this PR, and make a ticket to fix the error handling in this crate, and we can later change the root entropy files to use protobuf

**Comment by garbageslam** [comment]
i think we merged sarah's tool to convert these files to other formats as needed anyways

**Comment by garbageslam** [comment]
I guess we haven't merged that yet -- anyways, I have approved https://github.com/mobilecoinfoundation/mobilecoin/pull/545 and i'm going to close this one

**Pull request closed without merging**